staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -182,6 +182,10 @@ func (d *DiffProgram) getCommand(args ...string) (string, exec.Cmd) {
 			for i := 1; i < len(diffCommand); i++ {
 				if isValidChar(diffCommand[i]) {
 					args = append(args, diffCommand[i])
+				} else {
+					klog.Warningf(
+						"KUBECTL_EXTERNAL_DIFF command argument \"%v\" is ignored due to illegal characters. "+
+						"Acceptable characters include letters, numbers, dash, and equal sign.", diffCommand[i])
```

Comment:
I believe we must use a const for `^[a-zA-Z0-9-=]+$` and use it in the warning message as in line 181. The idea is avoid someone adding new item in the regex avoid to update the message above.